### PR TITLE
fix: page view count mismatch

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [x] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @llmin on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.60|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|27|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|19|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1325)
###### _updated: 5/17/2024, 08:03:36 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate failed

failed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1325&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1325&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1325&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @llmin on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.60|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|27|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:exclamation: Error|19|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1325)
###### _updated: 5/17/2024, 08:05:12 (UTC+0)_

File: src/main.ts [hunk 1]
```diff
@@ -18,20 +18,20 @@ import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
 import { BootstraplessStackSynthesizer, CompositeECRRepositoryAspect } from 'cdk-bootstrapless-synthesizer';
 import { AwsSolutionsChecks, NagPackSuppression, NagSuppressions } from 'cdk-nag';
 import { IConstruct, Node } from 'constructs';
-import { ApplicationLoadBalancerControlPlaneStack } from './alb-control-plane-stack';
-import { CloudFrontControlPlaneStack } from './cloudfront-control-plane-stack';
+// import { ApplicationLoadBalancerControlPlaneStack } from './alb-control-plane-stack';
```
Comment (zxkane) [inline src/main.ts:21]: revert unwanted changes

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 2]
```diff
@@ -1102,7 +1103,7 @@ const createOrUpdateRefreshSchedule = async (quickSight: QuickSight, commonParam
         logger.info('RefreshProperties exist, skip put operation.');
       } else {
         logger.error(`Put QuickSight refresh properties failed due to: ${(err as Error).message}`);
-        throw err;
+        // throw err;
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:1106]: don't ignore the exception

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 3]
```diff
@@ -1113,18 +1114,32 @@ const createOrUpdateRefreshSchedule = async (quickSight: QuickSight, commonParam
       datasetId,
       scheduleId,
     });
-    await quickSight.createRefreshSchedule({
-      AwsAccountId: commonParams.awsAccountId,
-      DataSetId: datasetId,
-      Schedule: {
+
+    let schedule: RefreshSchedule;
+    if (refreshInterval === RefreshInterval.HOURLY) {
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:1119]: assign a variable to  `RefreshInterval.DAILY` or `HOURLY`, then share the code to create schedule

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @llmin on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.60|< 80|
|Duplicated lines density|:white_check_mark: OK|5.20|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|8|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1325)
###### _updated: 5/17/2024, 12:33:40 (UTC+0)_

**Review by zxkane — APPROVED** [review]

**Review by tyyzqmf — APPROVED** [review]

**Pull request merged**